use_cluster.py

```diff
@@ -366,4 +355,391 @@ def return_cluster_dict(q_cluster, q_ids_list):
         else:
             final_clusters[q_cluster[indx]].append(q_ids_list[indx])
 
-    return final_clusters
\ No newline at end of file
+    return final_clusters
+
+
+"""
+> The objective of this function is to generate labels for each cluster based on a custom score developed by Araz, 
+using Normalised Mutual Information (NMI) and Distance from the Cluster Centroid 
+:The inputs required are the embeddings (generated previously from cluster functions), list of questions, question ids & clusters generated 
+: Read about NMI here: https://arxiv.org/pdf/1702.08199.pdf
+: For how Araz developed the custom score/metric, refer here: 
+"""
+
+def return_cluster_labels_NMI_nGrams_Centroid(embeddings,qs_list,q_ids_list,clusters): 
+
+    labelling_corpus = LabellingClusterCorpus(qs_list, nlp)
+    labelling_corpus.process_docs(embeddings, q_ids_list, clusters)
```

Comment:
I think we should call process_docs automatically in the LabellingClusterCorpus class's constructor, just as we do in the existing Corpus class. I don't think there's any time we'd want to create a LabellingClusterCorpus instance without processing the docs.

Reply:
I do still think its worth doing - there's no circumstance under which we'd want to create a LabellingClusterCorpus without processing the docs, so I think it should happen automatically without having to call process_docs separately.

I do agree it'd be better to reduce the number of parameters required for the constructor, though! One way to do this would be to use different data structures throughout the code. For example, instead of passing around q_ids_list, you could pass around a dictionary "questions" of the form:

{ questionId: questionDataDictionary }

Where questionDataDictionary is a dictionary with values like

{ embedding: embedding_for_that_question }

(Or even just the embedding itself instead of a dictionary if you don't need any other data with the questions).

Then wherever you have pass this dictionary, you can iterate through the questionIds by iterating through the dictionary keys, and you will automatically have the associated embedding when you do so.

Reply:
So Jeff, I thought about this, and I realised that after creating such a datastructure, we'll only be saving 1 parameter, and will still need to pass 4 instead of 5 while creating the instance. Further, I'll have to re-write the code inside the process docs function to use this, and it doesn't seem worth it just to save 1 parameter.

So for now I'm passing 5 parameters and calling the process docs inside the __init__ in the class :D I know it's not a good practice to have a lot of arguments, but I think for one initialisation, it seems ok. 

I'll push the new changes, and you can look and lemme know if it seems ok :D